### ISSUE TITLE
Can't init additional scopes

### ISSUE THREAD

**Issue by PolestarWorkshop** [googlesamples/google-signin-unity#59]
Hi, I can't pass additional scopes via GoogleSignIn.Configuration in Android - application crashed. What am I doing wrong ?

```
List<string> scopes = new List<string> { "https://www.googleapis.com/auth/contacts" };

        GoogleSignIn.Configuration = new GoogleSignInConfiguration
        {
            RequestAuthCode = true,
            RequestIdToken = true,
            AdditionalScopes = scopes,
            UseGameSignIn = false,
            WebClientId = "MY CLIENT ID"
        };

Task<GoogleSignInUser> signIn = GoogleSignIn.DefaultInstance.SignIn();
```

Application crashed with descibed below logs:

```
09-11 21:12:29.620: E/CRASH(2212): signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000004
09-11 21:12:29.620: E/CRASH(2212): *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
09-11 21:12:29.620: E/CRASH(2212): Build fingerprint: 'samsung/ks01ltexx/ks01ltexx:4.4.2/NJH47F/381180523:user/release-keys'
09-11 21:12:29.620: E/CRASH(2212): Revision: '0'
09-11 21:12:29.620: E/CRASH(2212): pid: 2212, tid: 2243, name: UnityMain  >>> com.sixgrid.BattleMagic <<<
09-11 21:12:29.620: E/CRASH(2212):     eax 00000004  ebx b4ea0c40  ecx 00000000  edx b92d18b4
09-11 21:12:29.620: E/CRASH(2212):     esi 00000004  edi 00000000
09-11 21:12:29.620: E/CRASH(2212):     xcs 00000073  xds 0000007b  xes 0000007b  xfs 00000000  xss 0000007b
09-11 21:12:29.620: E/CRASH(2212):     eip b4d3f7a0  ebp 7e9d6fd8  esp 7e9d6ef0  flags 00210292
09-11 21:12:29.620: E/CRASH(2212): backtrace:
09-11 21:12:29.620: E/CRASH(2212): 	#00  pc 000b47a0  /system/lib/libdvm.so (dvmCreateStringFromCstr(char const*)+32)
09-11 21:12:29.620: E/CRASH(2212): 	#01  pc 0000724b  /data/app-lib/com.sixgrid.BattleMagic-1/libnative-googlesignin.so (googlesignin::GoogleSignIn::SignIn()+59)
09-11 21:12:29.620: E/CRASH(2212): 	#02  pc 000063e3  /data/app-lib/com.sixgrid.BattleMagic-1/libnative-googlesignin.so (GoogleSignIn_SignIn+51)
09-11 21:12:29.620: E/CRASH(2212): 	#03  pc 000068af   ( (wrapper managed-to-native) Google.Impl.GoogleSignInImpl:GoogleSignIn_SignIn (System.Runtime.InteropServices.HandleRef) + 0x2f (0x774c4880 0x774c48d4) [0xd0ee0 - Unity Root Domain]+26799)
09-11 21:12:29.620: E/CRASH(2212): 	#04  pc 00006820   ( Google.Impl.GoogleSignInImpl:SignIn () + 0x30 (0x774c47f0 0x774c4861) [0xd0ee0 - Unity Root Domain]+26656)
09-11 21:12:29.620: E/CRASH(2212): 	#05  pc 000066ea   ( Google.GoogleSignIn:SignIn () + 0x4a (0x774c46a0 0x774c4722) [0xd0ee0 - Unity Root Domain]+26346)
09-11 21:12:29.620: E/CRASH(2212): 	#06  pc 000051bd   ( GooglePlusButton:connect () + 0x95 (0x774c3128 0x774c3243) [0xd0ee0 - Unity Root Domain]+20925)
09-11 21:12:29.620: E/CRASH(2212): 	#07  pc 000050b3   ( GooglePlusButton:click () + 0x33 (0x774c3080 0x774c30b8) [0xd0ee0 - Unity Root Domain]+20659)
09-11 21:12:29.620: E/CRASH(2212): 	#08  pc 0000097c   ( UnityEngine.Events.InvokableCall:Invoke () + 0x2c (0x774be950 0x774be981) [0xd0ee0 - Unity Root Domain]+2428)
09-11 21:12:29.620: E/CRASH(2212): 	#09  pc 0000f31a   ( UnityEngine.Events.UnityEvent:Invoke () + 0x6a (0x774272b0 0x774273ea) [0xd0ee0 - Unity Root Domain]+62234)
09-11 21:12:29.620: E/CRASH(2212): 	#10  pc 0000f295   ( UnityEngine.UI.Button:Press () + 0x55 (0x77427240 0x7742729e) [0xd0ee0 - Unity Root Domain]+62101)
09-11 21:12:29.620: E/CRASH(2212): 	#11  pc 0000f228   ( UnityEngine.UI.Button:OnPointerClick (UnityEngine.EventSystems.PointerEventData) + 0x28 (0x77427200 0x7742722d) [0xd0ee0 - Unity Root Domain]+61992)
09-11 21:12:29.620: E/CRASH(2212): 	#12  pc 0000f1f8   ( UnityEngine.EventSystems.ExecuteEvents:Execute (UnityEngine.EventSystems.IPointerClickHandler,UnityEngine.EventSystems.BaseEventData) + 0x30 (0x774271c8 0x774271fd) [0xd0ee0 - Unity Root Domain]+61944)
09-11 21:12:29.620: E/CRASH(2212): 	#13  pc 00005a30   ( UnityEngine.EventSystems.ExecuteEvents:Execute<object> (UnityEngine.GameObject,UnityEngine.EventSystems.BaseEventData,UnityEngine.EventSystems.ExecuteEvents/EventFunction`1<object>) + 0x1d8 (0x78244858 0x78244af0) [0xd0ee0 - Unity Root Domain]+23088)
09-11 21:12:29.620: E/CRASH(2212): 	#14  pc 00004589   ( UnityEngine.EventSystems.StandaloneInputModule:ProcessTouchPress (UnityEngine.EventSystems.PointerEventData,bool,bool) + 0x3c1 (0x782431c8 0x782436c5) [0xd0ee0 - Unity Root Domain]+17801)
09-11 21:12:29.620: E/CRASH(2212): 	#15  pc 00007aaa   ( UnityEngine.EventSystems.StandaloneInputModule:ProcessTouchEvents () + 0xe2 (0x79d559c8 0x79d55b42) [0xd0ee0 - Unity Root Domain]+31402)
09-11 21:12:29.620: E/CRASH(2212): 	#16  pc 000072e1   ( UnityEngine.EventSystems.StandaloneInputModule:Process () + 0x91 (0x79d55250 0x79d5531d) [0xd0ee0 - Unity Root Domain]+29409)
09-11 21:12:29.620: E/CRASH(2212): 	#17  pc 0000b5f9   ( UnityEngine.EventSystems.EventSystem:Update () + 0x171 (0x77ecf488 0x77ecf604) [0xd0ee0 - Unity Root Domain]+46585)
09-11 21:12:29.620: E/CRASH(2212): 	#18  pc 000070e9   ( (wrapper runtime-invoke) object:runtime_invoke_void__this__ (object,intptr,intptr,intptr) + 0x41 (0x7f92d0a8 0x7f92d13d) [0xd0ee0 - Unity Root Domain]+28905)
09-11 21:12:29.620: E/CRASH(2212): 	#19  pc 0002144f  /data/app-lib/com.sixgrid.BattleMagic-1/libmono.so
09-11 21:12:29.620: E/CRASH(2212): 	#20  pc 001d033b  /data/app-lib/com.sixgrid.BattleMagic-1/libmono.so (mono_runtime_invoke+141)
09-11 21:12:29.620: E/CRASH(2212): 	#21  pc 004aa9df  /data/app-lib/com.sixgrid.BattleMagic-1/libunity.so
09-11 21:12:29.620: E/CRASH(2212): 	#22  il 00000004  at (wrapper managed-to-native) Google.Impl.GoogleSignInImpl.GoogleSignIn_SignIn (System.Runtime.InteropServices.HandleRef) <0x00004>
09-11 21:12:29.620: E/CRASH(2212): 	#23  il 00000004  at (wrapper managed-to-native) Google.Impl.GoogleSignInImpl.GoogleSignIn_SignIn (System.Runtime.InteropServices.HandleRef) <0x00004>
09-11 21:12:29.620: E/CRASH(2212): 	#24  il 0000002f  at Google.Impl.GoogleSignInImpl.SignIn () <0x0002f>
09-11 21:12:29.620: E/CRASH(2212): 	#25  il 00000049  at Google.GoogleSignIn.SignIn () <0x00049>
09-11 21:12:29.620: E/CRASH(2212): 	#26  il 00000094  at GooglePlusButton.connect () <0x00094>
09-11 21:12:29.620: E/CRASH(2212): 	#27  il 00000032  at GooglePlusButton.click () <0x00032>
09-11 21:12:29.620: E/CRASH(2212): 	#28  il 0000002b  at UnityEngine.Events.InvokableCall.Invoke () <0x0002b>
09-11 21:12:29.620: E/CRASH(2212): 	#29  il 00000069  at UnityEngine.Events.UnityEvent.Invoke () <0x00069>
09-11 21:12:29.620: E/CRASH(2212): 	#30  il 00000054  at UnityEngine.UI.Button.Press () <0x00054>
09-11 21:12:29.620: E/CRASH(2212): 	#31  il 00000027  at UnityEngine.UI.Button.OnPointerClick (UnityEngine.EventSystems.PointerEventData) <0x00027>
09-11 21:12:29.620: E/CRASH(2212): 	#32  il 0000002f  at UnityEngine.EventSystems.ExecuteEvents.Execute (UnityEngine.EventSystems.IPointerClickHandler,UnityEngine.EventSystems.BaseEventData) <0x0002f>
09-11 21:12:29.620: E/CRASH(2212): 	#33  il 000001d7  at UnityEngine.EventSystems.ExecuteEvents.Execute<object> (UnityEngine.GameObject,UnityEngine.EventSystems.BaseEventData,UnityEngine.EventSystems.ExecuteEvents/EventFunction`1<object>) <0x001d7>
09-11 21:12:29.620: E/CRASH(2212): 	#34  il 000003c0  at UnityEngine.EventSystems.StandaloneInputModule.ProcessTouchPress (UnityEngine.EventSystems.PointerEventData,bool,bool) <0x003c0>
09-11 21:12:29.620: E/CRASH(2212): 	#35  il 000000e1  at UnityEngine.EventSystems.StandaloneInputModule.ProcessTouchEvents () <0x000e1>
09-11 21:12:29.620: E/CRASH(2212): 	#36  il 00000090  at UnityEngine.EventSystems.StandaloneInputModule.Process () <0x00090>
09-11 21:12:29.620: E/CRASH(2212): 	#37  il 00000170  at UnityEngine.EventSystems.EventSystem.Update () <0x00170>
09-11 21:12:29.620: E/CRASH(2212): 	#38  il 00000040  at (wrapper runtime-invoke) object.runtime_invoke_void__this__ (object,intptr,intptr,intptr) <0x00040>
09-11 21:12:29.620: E/CRASH(2212): 	#00  pc 0002144e  /data/app-lib/com.sixgrid.BattleMagic-1/libmono.so
09-11 21:12:29.620: E/CRASH(2212): 	#01  pc 001d033b  /data/app-lib/com.sixgrid.BattleMagic-1/libmono.so (mono_runtime_invoke+141)
09-11 21:12:29.620: E/CRASH(2212): 	#02  pc 004aa9df  /data/app-lib/com.sixgrid.BattleMagic-1/libunity.so
09-11 21:12:29.630: A/MessageQueue(2212): IdleHandler threw exception
09-11 21:12:29.630: A/MessageQueue(2212): java.lang.Error: signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 00000004
09-11 21:12:29.630: A/MessageQueue(2212): Build fingerprint: 'samsung/ks01ltexx/ks01ltexx:4.4.2/NJH47F/381180523:user/release-keys'
09-11 21:12:29.630: A/MessageQueue(2212): Revision: '0'
09-11 21:12:29.630: A/MessageQueue(2212): pid: 2212, tid: 2243, name: UnityMain  >>> com.sixgrid.BattleMagic <<<
09-11 21:12:29.630: A/MessageQueue(2212): 	at libdvm.dvmCreateStringFromCstr(char const*)(dvmCreateStringFromCstr:32)
09-11 21:12:29.630: A/MessageQueue(2212): 	at libnative-googlesignin.googlesignin::GoogleSignIn::SignIn()(GoogleSignIn:59)
09-11 21:12:29.630: A/MessageQueue(2212): 	at libnative-googlesignin.GoogleSignIn_SignIn(GoogleSignIn_SignIn:51)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000068af(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.00006820(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000066ea(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000051bd(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000050b3(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000097c(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000f31a(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000f295(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000f228(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000f1f8(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.00005a30(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.00004589(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.00007aaa(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000072e1(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.0000b5f9(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Unknown.000070e9(Unknown Source)
09-11 21:12:29.630: A/MessageQueue(2212): 	at libmono.0002144f(Native Method)
09-11 21:12:29.630: A/MessageQueue(2212): 	at libmono.mono_runtime_invoke(mono_runtime_invoke:141)
09-11 21:12:29.630: A/MessageQueue(2212): 	at libunity.004aa9df(Native Method)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GoogleSignInImpl.GoogleSignIn_SignIn(Native Method)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GoogleSignInImpl.GoogleSignIn_SignIn(Native Method)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GoogleSignInImpl.SignIn(<0x0002f>:47)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GoogleSignIn.SignIn(<0x00049>:73)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GooglePlusButton.connect(<0x00094>:148)
09-11 21:12:29.630: A/MessageQueue(2212): 	at GooglePlusButton.click(<0x00032>:50)
09-11 21:12:29.630: A/MessageQueue(2212): 	at InvokableCall.Invoke(<0x0002b>:43)
09-11 21:12:29.630: A/MessageQueue(2212): 	at UnityEvent.Invoke(<0x00069>:105)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Button.Press(<0x00054>:84)
09-11 21:12:29.630: A/MessageQueue(2212): 	at Button.OnPointerClick(<0x00027>:39)

```

**Comment by claywilkinson** [comment]
I believe this is fixed in version 1.0.4

**Comment by PolestarWorkshop** [comment]
> I believe this is fixed in version 1.0.4

It's working fine now! Thanks, issue closed.
